state/application.go

```diff
@@ -474,6 +484,56 @@ func (op *DestroyApplicationOperation) destroyOps() ([]txn.Op, error) {
 	return ops, nil
 }
 
+func (op *DestroyApplicationOperation) unassignBranchOps() ([]txn.Op, error) {
+	m, err := op.app.st.Model()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	appName := op.app.doc.Name
+	branches, err := m.applicationBranches(appName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(branches) == 0 {
+		return nil, errors.NotFoundf("branch for %q", appName)
+	}
+	ops := []txn.Op{}
+	for _, branch := range branches {
```

Comment:
I would consider making this logic (and come to think of it, the remove unit ops as well) a member of the branch itself for nice separation of concerns.

Reply:
done for this logic, in unassignUnitFromBranch in pr 10415